code/ai/aicode.cpp

```diff
@@ -9210,10 +9245,20 @@ float dock_orient_and_approach(object *docker_objp, int docker_index, object *do
 		if (sip0->flags[Ship::Info_Flags::Support])
 			vm_vec_scale(&acc_limit, 2.0f);
 
+		if(Framerate_independent_turning)
+			ai_compensate_for_retail_turning(&vel_limit, &acc_limit, docker_objp->phys_info.rotdamp, false);
+
 		// true at end of line prevent overshoot
-		vm_angular_move_matrix(&dom, &docker_objp->orient, &omega_in, flFrametime, &nm, &omega_out, &vel_limit, &acc_limit, false, true);
-		docker_objp->phys_info.rotvel = omega_out;
-		docker_objp->orient = nm;
+		vm_angular_move_matrix(&dom, &docker_objp->orient, &omega_in, flFrametime, &out_orient, &omega_out, &vel_limit, &acc_limit, false, true);
+
+		if (Framerate_independent_turning) {
```

Comment:
You had an opportunity, too, to test docker behavior?

Reply:
I didn't test it explicitly, although I've rearmed myself quite a few times during my testing. I'll do some more explicit tests to be sure.

Reply:
Yep, everything seems to be in order.